spec/controllers/animals_ controller_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'spec_helper'
+
+describe Api::V1::AnimalsController do
+  let!(:user) { FactoryGirl.create(:user, account_active: 'true', permissions: 'animals_edit') }
+  let!(:animal) { FactoryGirl.create(:animal) }
+   
+  before(:each) do
+    @request.env['devise.mapping'] = Devise.mappings[:user]
```

Comment:
Es lo que permite que pueda hacer sign in por ejemplo, me mapea las rutas del usuario para los tests